src/transform/WidenImage.js

```diff
@@ -1,30 +1,99 @@
+    if (el.classList.contains('content_block')) {
+      break
+    }
+    widenThese.push(el)
+  }
+  return widenThese
+}
+
+/**
+ * Sets style value.
+ * @param {HTMLElement.style} style
+ * @param {!string} key
+ * @param {*} value
+ * @return {void}
+ */
+const updateStyleValue = (style, key, value) => {
+  style[key] = value
+}
+
+/**
+ * Sets style value only if value for given key already exists.
+ * @param {HTMLElement.style} style
+ * @param {!string} key
+ * @param {*} value
+ * @return {void}
+ */
+const updateExistingStyleValue = (style, key, value) => {
+  const existingValue = style[key]
+  if (!existingValue) {
+    return
+  }
+  updateStyleValue(style, key, value)
+}
+
+/**
+ * Array of arrays of image widening CSS key/value pairs.
+ * @type {Array}
+ */
+const styleWideningKeysAndValues = [
+  ['width', '100%'],
+  ['height', 'auto'],
+  ['maxWidth', '100%'],
+  ['float', 'none']
+]
+
+/**
+ * Perform widening on an element. Certain style properties are updated, but only if existing values
+ * for these properties already exist.
+ * @param  {!HTMLElement} element
+ * @return {void}
+ */
+const widenElementByUpdatingExistingStyles = element => {
+  styleWideningKeysAndValues
+    .forEach(keyAndValue => updateExistingStyleValue(element.style, ...keyAndValue))
+}
+
+/**
+ * Perform widening on an element.
+ * @param  {!HTMLElement} element
+ * @return {void}
+ */
+const widenElementByUpdatingStyles = element => {
+  styleWideningKeysAndValues
+    .forEach(keyAndValue => updateStyleValue(element.style, ...keyAndValue))
+}
+
 /**
  * To widen an image element a css class called 'pagelib_widen_image_override' is applied to the
  * image element, however, ancestors of the image element can prevent the widening from taking
  * effect. This method makes minimal adjustments to ancestors of the image element being widened so
  * the image widening can take effect.
- * @param  {!HTMLElement} el Element whose ancestors will be widened
+ * @param  {!HTMLElement} element Element whose ancestors will be widened
  * @return {void}
  */
-const widenAncestors = el => {
-  for (let parentElement = el.parentElement;
-    parentElement && !parentElement.classList.contains('content_block');
-    parentElement = parentElement.parentElement) {
-    if (parentElement.style.width) {
-      parentElement.style.width = '100%'
-    }
-    if (parentElement.style.height) {
-      parentElement.style.height = 'auto'
-    }
-    if (parentElement.style.maxWidth) {
-      parentElement.style.maxWidth = '100%'
-    }
-    if (parentElement.style.float) {
-      parentElement.style.float = 'none'
-    }
+const widenAncestors = element => {
+  ancestorsToWiden(element).forEach(widenElementByUpdatingExistingStyles)
+
+  // Without forcing widening on the parent anchor, lazy image loading placeholders
+  // aren't correctly widened on iOS for some reason.
+  const parentAnchor = elementUtilities.findClosestAncestor(element, 'a.image')
```

Comment:
Should we just check if element.parentElement is an instanceof HTMLAnchorElement instead?

Reply:
I had that initially at first... but I wasn't sure if there were ever cases where some sort of template delivered img wrapped with anchor with some filling in between... I'm fine with switching back if you think its ok...